### DISCUSSION TITLE
我想获取在 .umirc.ts中配置的router信息，有什么办法吗？

### DISCUSSION THREAD

**Discussion by balabll** [umijs/umi#11441] [Q&A] [answered]
我想获取在 .umirc.ts中配置的router信息，有什么办法吗？

**Comment by BoyYangzai** [comment]
> 我想获取在 .umirc.ts中配置的router信息，有什么办法吗？

v3还是v4呢

**Reply by balabll** [reply]
@BoyYangzai 我们既有使用3.x的项目，也有使用4.x的项目，目前是想封装一个公共组件，这个组件中期望能够获取到在 `umirc.ts`配置的router信息

**Comment by fz6m** [comment, marked as answer]
最直接的方法就是把 `routes` 单独配置到一个文件里，然后 `.umirc.ts` 和项目里的代码分别导入它，这样就可以公用了。

或者，可以看看 [`useAppData`](https://umijs.org/docs/api/api#useappdata) 里有没有你需要的数据。

**Comment by balabll** [comment]
> 最直接的方法就是把 `routes` 单独配置到一个文件里，然后 `.umirc.ts` 和项目里的代码分别导入它，这样就可以公用了。
> 
> 或者，可以看看 [`useAppData`](https://umijs.org/docs/api/api#useappdata) 里有没有你需要的数据。

@fz6m 感谢，我试一下 `useAppData`
